scalding-spark/src/test/scala/com/twitter/scalding/spark_backend/SparkBackendTests.scala

```diff
@@ -67,6 +67,12 @@ class SparkBackendTests extends FunSuite with BeforeAndAfter {
       t.toIterableExecution.waitFor(Config.empty, MemoryMode.empty).get
     )
 
+  def sparkRetrieveCounters[A: Ordering](t: TypedPipe[A], conf: Config = Config.empty) = {
+    val smode = SparkMode.default(session)
+    val (eiter, ecounters) = t.toIterableExecution.getCounters.waitFor(conf, smode).get
```

Comment:
I believe the call to toLocalIterator happens after the `execute` call for `SparkWriter`. So I'm not sure how we can lift that call for `execute` to be aware of it. 

But... that would definitely be the cleanest solution to use instead of the `count` hack I currently have